.github/workflows/build.yml

```diff
@@ -0,0 +1,95 @@
+name: Validate Build
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  docker-validate:
+    name: Validate Build
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: docker.io/library/registry
+        ports:
+          - 5000:5000
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.24
+      - uses: arduino/setup-protoc@v3
+        with:
+          version: 31.1
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install Tools
+        run: |
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.36
+          go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.5
+          go install github.com/matryer/moq@v0.5
+      - name: Install Dependencies
+        run: go get ./...
+      - name: Generate Files
+        run: |
+          go generate ./...
+      - name: Build Binary
+        run: |
```

Comment:
Just as a random aside, it feels risky that this GitHub Action to validate things is not actually running the same build script that the production builds will run. It's replicating (I think?) the steps in the Makefile, which means that if those steps change in future, this validation action could fall out of sync. Shouldn't it just run `make dist` here?